adr/ADR-0009-project-workflow.md

```diff
@@ -0,0 +1,623 @@
+    - [Python TrustyAI](https://github.com/trustyai-explainability/trustyai-explainability-python)
+    - Llama Stack out-of-tree providers
+        - [LMEval](https://github.com/trustyai-explainability/llama-stack-provider-lmeval)
+        - [Guardrails](https://github.com/trustyai-explainability/llama-stack-provider-trustyai-fms)
+
+2. **Midstream** (https://github.com/opendatahub-io) - forks of upstream repositories
+
+3. **Downstream** - forks of midstream repositories
+
+The current development process presents several challenges:
+- Manual handling of code propagation between repositories
+- Complex development cycles
+- Difficulty controlling which features propagate downstream
+- Maintaining reference copies of upstream code
+- Allowing experimentation without breaking downstream
+- Minimising the number of branches in use
+
+## Goals
+
+- Streamline Git workflow across upstream, midstream, and downstream repositories
+- Minimise the number of branches in use
+- Simplify development cycles
+- Ensure all development and experimentation happens upstream only
+- Support incubation and testing of upstream features in ODH before downstream release
+- Provide precise control over feature propagation to downstream
+- Automate container image building and `params.env` updates
+
+## Non-goals
+
+- Creating completely separate development paths for each repository
+- Maintaining multiple parallel long-term feature branches
+- Fully automating all aspects of the development workflow without human oversight
+- Developing or experimenting with features in midstream (ODH) repositories
+
+## Current situation
+
+The project currently lacks a standardised workflow across repositories, resulting in manual code propagation, complex development cycles, and difficulty controlling feature propagation downstream.
+
+## Proposal
+
+We will implement a streamlined Git workflow with the following structure:
+
+### Repository Structure
+- **Upstream**: Single persistent branch: `main` - Used for all development and experimentation
+- **Midstream**: Three persistent branches:
+    - `main` - Complete mirror of **upstream:main**
+    - `incubation` - Branch for ODH feature testing, validation, and releases (no development)
+    - `stable` - Curated set of validated features for downstream consumption
+- **Downstream**: Single persistent branch: `main` - Syncs from **midstream:stable**
+- Temporary feature branches: created via PRs and deleted after merge
+- Release branches: created from **midstream:incubation** for ODH releases (_e.g._, **odh/2.20**, **odh/2.21**)
+
+### Release Cycle
+
+The project components will follow these release cadences:
+- **Upstream**: Releases every 3 weeks
+- **Midstream (ODH)**: Releases every 3 weeks, coinciding with upstream releases
+- **Downstream**: Monthly releases
+
+This schedule allows upstream features to be developed, tested in ODH, and then propagated to downstream after a period of incubation.
+
+### Branch Flow
```

Comment:
i am thinking:
ODH release branch is out of ODH incubation
ODH stable branch is out of ODH incubation as well ( for simplicity, lets say ODH stable == downstream main)
due to different release cycles (3 weeks VS monthly-ish) , wont there be a chance, a commit/feature on ODH incubation has not been sync to ODH release (out as part of ODH release) but get onto  ODH stable (in downstream release)?